switch/ffmpeg/tls.patch

```diff
@@ -0,0 +1,276 @@
+new file mode 100644
+index 0000000..c13c1d2
+--- /dev/null
++++ b/libavformat/tls_libnx.c
+@@ -0,0 +1,186 @@
++/*
++ * TLS/SSL Protocol
++ * Copyright (c) 2018 Thomas Volkert
++ *
++ * This file is part of FFmpeg.
++ *
++ * FFmpeg is free software; you can redistribute it and/or
++ * modify it under the terms of the GNU Lesser General Public
++ * License as published by the Free Software Foundation; either
++ * version 2.1 of the License, or (at your option) any later version.
++ *
++ * FFmpeg is distributed in the hope that it will be useful,
++ * but WITHOUT ANY WARRANTY; without even the implied warranty of
++ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
++ * Lesser General Public License for more details.
++ *
++ * You should have received a copy of the GNU Lesser General Public
++ * License along with FFmpeg; if not, write to the Free Software
++ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
++ */
++
++#include <switch.h>
++
++#include "avformat.h"
++#include "internal.h"
++#include "url.h"
++#include "tls.h"
++#include "libavutil/parseutils.h"
++
++typedef struct TLSContext {
++    const AVClass *class;
++    TLSShared tls_shared;
++    SslContext context;
++    SslConnection conn;
++} TLSContext;
++
++static int tls_close(URLContext *h)
++{
++    TLSContext *tls_ctx = h->priv_data;
++
++    sslConnectionClose(&tls_ctx->conn);
++    sslContextClose(&tls_ctx->context);
++
++    ffurl_closep(&tls_ctx->tls_shared.tcp);
++    return 0;
++}
++
++static int tls_open(URLContext *h, const char *uri, int flags, AVDictionary **options)
++{
++    TLSContext *tls_ctx = h->priv_data;
++    TLSShared *shr = &tls_ctx->tls_shared;
++    int ret;
++    Result rc;
++
++    if ((ret = ff_tls_open_underlying(shr, h, uri, options)) < 0)
++        goto fail;
++
++    if (shr->listen) {
++        av_log(h, AV_LOG_ERROR, "TLS Listen Sockets with libnx is not implemented.\n");
++        ret = AVERROR(EINVAL);
++        goto fail;
++    }
++
++    rc = sslCreateContext(&tls_ctx->context, SslVersion_Auto);
++    if (R_FAILED(rc)) {
++        ret = AVERROR(EINVAL);
++        goto fail;
++    }
++
++    rc = sslContextCreateConnection(&tls_ctx->context, &tls_ctx->conn);
++
++    if (R_SUCCEEDED(rc)) {
++        rc = sslConnectionSetOption(&tls_ctx->conn, SslOptionType_DoNotCloseSocket, true);
++    }
++
++    if (R_SUCCEEDED(rc)) {
++        int *sockfd = shr->tcp->priv_data + sizeof(const AVClass *);
++        ret = socketSslConnectionSetSocketDescriptor(&tls_ctx->conn, *sockfd);
++        if (ret == -1 && errno != ENOENT) {
++            av_log(h, AV_LOG_ERROR, "socketSslConnectionSetSocketDescriptor returned: %u\n", rc);
++            goto fail;
++        }
++    }
++
++    if (R_SUCCEEDED(rc)) {
++        rc = sslConnectionSetHostName(&tls_ctx->conn, shr->host, strlen(shr->host));
++    }
++
++    /* This will fail on system-versions where this option isn't available,
```

Comment:
Is this an outdated comment? You don't appear to be ignoring errors here.

Reply:
fixed error handling